src/sonic-frr/Makefile

```diff
@@ -5,14 +5,55 @@ SHELL = /bin/bash
 MAIN_TARGET = $(FRR)
 DERIVED_TARGET = $(FRR_PYTHONTOOLS) $(FRR_DBG) $(FRR_SNMP) $(FRR_SNMP_DBG)
 
+patch1 = 0001-Add-support-of-bgp-tcp-DSCP-value.patch
+patch2 = 0002-Reduce-severity-of-Vty-connected-from-message.patch
+patch3 = 0003-ignore-nexthop-attribute-when-NLRI-is-present.patch
+patch4 = 0004-Allow-BGP-attr-NEXT_HOP-to-be-0.0.0.0-due-to-allevia.patch
+patch5 = 0005-Support-VRF.patch
+
 $(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
 	# Build the package
 	pushd ./frr
-	patch -p1 < ../patch/0001-Add-support-of-bgp-tcp-DSCP-value.patch
-	patch -p1 < ../patch/0002-Reduce-severity-of-Vty-connected-from-message.patch
-	patch -p1 < ../patch/0003-ignore-nexthop-attribute-when-NLRI-is-present.patch
-	patch -p1 < ../patch/0004-Allow-BGP-attr-NEXT_HOP-to-be-0.0.0.0-due-to-allevia.patch
-	patch -p1 < ../patch/0005-Support-VRF.patch
+	if patch -sfp1 --dry-run < ../patch/$(patch1); then
+		patch -sfp1 < ../patch/$(patch1)
+	else
+		if ! patch -Rsfp1 --dry-run < ../patch/$(patch1); then
+			echo "Patch $(patch1) failure"
+			exit 1
+		fi
+	fi
+	if patch -sfp1 --dry-run < ../patch/$(patch2); then
+		patch -sfp1 < ../patch/$(patch2)
+	else
+		if ! patch -Rsfp1 --dry-run < ../patch/$(patch2); then
+			echo "Patch $(patch2) failure"
+			exit 1
+		fi
+	fi
+	if patch -sfp1 --dry-run < ../patch/$(patch3); then
+		patch -sfp1 < ../patch/$(patch3)
+	else
+		if ! patch -Rsfp1 --dry-run < ../patch/$(patch3); then
+			echo "Patch $(patch3) failure"
+			exit 1
+		fi
+	fi
+	if patch -sfp1 --dry-run < ../patch/$(patch4); then
+		patch -sfp1 < ../patch/$(patch4)
+	else
+		if ! patch -Rsfp1 --dry-run < ../patch/$(patch4); then
+			echo "Patch $(patch4) failure"
+			exit 1
+		fi
+	fi
+	if patch -sfp1 --dry-run < ../patch/$(patch5); then
```

Comment:
Hi,
what does "if patch -sfp1 --dry-run < ../patch/$(patch5);" mean ? Also, I am adding a new patch6. For every new patch, should we change the sonic-frr/makefile.

Reply:
Yes, with each patch the makefile would have to be changed. `if patch -sfp1 --dry-run < ../patch/$(patch5)` tries (without applying hence the dry-run) to see if the patch can apply.